### PR TITLE
Restructure how CLI arguments are read for qmcfinitesize

### PR DESCRIPTION
**Pull request by mcbennet**
## Proposed changes

Minor change, however, helpful for new users running the tool.
The usage is printed when number of arguments is less than what is needed for successful execution.
Also fixes issue where 'qmcfinitesize --help' prints nothing.

## What type(s) of changes does this code introduce?

- Other (please describe): Minor "fix". Before this PR 'qmcfinitesize --help' leads to no printout

### Does this introduce a breaking change?

- No

## What systems has this change been tested on?

Desktop

## Checklist

_Update the following with a yes where the items apply. If you're unsure about any of them, don't hesitate to ask.  This is
simply a reminder of what we are going to look for before merging your code._

- Yes. This PR is up to date with current the current state of 'develop'
- No. Code added or changed in the PR has been clang-formatted
- No. This PR adds tests to cover any new code, or to catch a bug that is being fixed
- No. Documentation has been added (if appropriate)

### CODE REVIEW

File: src/QMCTools/qmcfinitesize.cpp [hunk 1]
```diff
@@ -57,39 +57,44 @@ int main(int argc, char** argv)
   std::cout.precision(12);
 
   std::unique_ptr<SkParserBase> skparser(nullptr);
-  int iargc = 2;
 
-  while (iargc + 1 < argc)
+  if (argc<4)
```
Comment (ye-luo) [inline src/QMCTools/qmcfinitesize.cpp:61]: Add a comment before this like to help understanding the code. Also clang-format the change if you can.

**Review by camelto2 — DISMISSED** [review]

File: src/QMCTools/qmcfinitesize.cpp (hunk 1 above)
Comment (mcbennet) [reply src/QMCTools/qmcfinitesize.cpp]: Addressed.

**Comment by mcbennet** [comment]
> Minor comment: There seems to be lots of ways to pass malformed or improper arguments to this tool.

@prckent To help remedy, I can add a few catches along with this PR

**Comment by ye-luo** [comment]
Okay to test

**Pull request merged**